main.go

```diff
@@ -121,8 +121,9 @@ func main() {
 	)
 
 	if err = (&apimcontrollers.RateLimitPolicyReconciler{
-		BaseReconciler: rateLimitPolicyBaseReconciler,
-		Scheme:         mgr.GetScheme(),
+		TargetRefReconciler: reconcilers.TargetRefReconciler{
+			BaseReconciler: rateLimitPolicyBaseReconciler,
+		},
```

Comment:
```suggestion
		TargetRefReconciler: reconcilers.NewTargetRefReconciler(rateLimitPolicyBaseReconciler),
```

Reply:
if there was anything to be done by the builder, then it would make sense. Currently, it is just a cosmetic change

Reply:
I see it as abstraction and a good practice rather than just cosmetic, but I guess the same argument could be used to propose also `NewRateLimitPolicyReconciler()` and `NewAuthPolicyReconciler()`. Not critical anyway.